docs/security-considerations.rst

```diff
@@ -344,3 +346,118 @@ Note that formal verification itself can only help you understand the
 difference between what you did (the specification) and how you did it
 (the actual implementation). You still need to check whether the specification
 is what you wanted and that you did not miss any unintended effects of it.
+
+Solidity implements a formal verification approach based on SMT solving.  The
+SMTChecker module automatically tries to prove that the code satisfies the
+specification given by ``require/assert`` statements, that is, it considers
+``require`` statements as assumptions and tries to prove that the conditions
+inside ``assert`` statements are always true.  If an assertion failure is
+found, a counterexample is given to the user, showing how the assertion can be
+violated.
+
+The SMTChecker also checks automatically for underflow/overflow, trivial
+conditions and unreachable code.
+It is currently an experimental feature, therefore in order to use it you need
+to enable it via :ref:`a pragma directive<smt_checker>`.
+
+The SMTChecker traverses the Solidity AST creating and collecting program constraints.
+When it encounters a verification target, an SMT solver is invoked to determine the outcome.
+
+For more details on how the SMT encoding works internally, see
+`our paper <https://github.com/leonardoalt/text/blob/master/solidity_isola_2018/main.pdf>`_.
```

Comment:
I would not use the word "our" here. Perhaps more neutral: "There is an article that describes..."

Reply:
Ok!
Ok to link my GH here? I can also point to Springer or smth

Reply:
The more open the better, I would say.